pkg/ipamd/ipamd.go

```diff
@@ -311,6 +312,17 @@ func New(k8sapiClient kubernetes.Interface, eniConfig *eniconfig.ENIConfigContro
 	if err != nil {
 		return nil, err
 	}
+
+	mac := c.awsClient.GetPrimaryENImac()
+	// retrieve security groups
+	err = c.awsClient.RefreshSGIDs(mac)
```

Comment:
Is this required? I'm assuming we're not going to update SGs on ENIs of Non-Zero NW cards anyways, right? Is this code block addressing some other known issue?

Reply:
Yeah, so this was called in NodeInit and this function gets attached ENIs and updates the SGs. Which again thinks the ENIs are secondary ENIs. So moved it after describe call since by then we would have built local cache on CNIUnmanaged ENIs.